### PR TITLE
[ENH] Add support to get_slice for multi-index and hierarchical data

### PR DESCRIPTION
**Pull request by bethrice44**
#### Reference Issues/PRs
Issue #2757 

#### What does this implement/fix? Explain your changes.

Adds support for multi-index and hierarchical data types to `get_slice` (based on `get_window`)

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

Are the assumptions for index types valid? I believe `get_window` is implicitly making the same assumptions.

If they are valid, can we amend the examples so series/ dfs have a timestamp index by default for testing?

### CODE REVIEW

File: sktime/datatypes/_utilities.py [hunk 1]
```diff
@@ -331,33 +333,35 @@ def get_slice(obj, start=None, end=None):
...
 
-    if (
-        isinstance(start, pd.Timestamp)
-        or isinstance(end, pd.Timestamp)
-        or isinstance(start, pd.Period)
-        or isinstance(end, pd.Period)
-    ):
+    # numpy3D (Panel) or np.npdarray (Series)
+    # Assumes the index is integer so will be exclusive by default
+    if isinstance(obj, np.ndarray):
```
Comment (fkiraly) [inline sktime/datatypes/_utilities.py:338]: hm, I think this is not quite right.

`numpy3D` (for a panel of series) has index (instance, variable, time), so you need to subset the *third* index in that case.
`np.ndarray` has (time, variable) indexing.

I know, this is strangely inconsistent, but it is individually consistent with community preference and the `tslearn` package for time series classification.

**Review by fkiraly — CHANGES_REQUESTED** [review]
I think this is good, but does not cover `numpy3D` correctly.

You need to cover the types in `GET_LATEST_WINDOW_SUPPORTED_MTYPES`, the others are dealt with by the conversion.

Out of these, I think `numpy3D` is being dealt with incorrectly, see above.

File: sktime/datatypes/_utilities.py (hunk 1 above)
Comment (bethrice44) [reply sktime/datatypes/_utilities.py:348]: Looks good

**Review by fkiraly — APPROVED** [review]
All fine now & thanks for the review of my fix.

**Comment by fkiraly** [comment]
What did you mean by this, btw?

> Are the assumptions for index types valid? I believe get_window is implicitly making the same assumptions.
> If they are valid, can we amend the examples so series/ dfs have a timestamp index by default for testing?

Which assumptions?

**Pull request merged**